ob-jq.el

```diff
@@ -52,49 +53,66 @@
+      (setq data (cdr data)))
+    (json-encode
+     (mapcar
+      (lambda (row) (cl-mapcar 'cons header row))
+      data))))
+
 (defun org-babel-execute:jq (body params)
   "Execute a block of jq code with org-babel.  This function is
 called by `org-babel-execute-src-block'"
   (message "executing jq source code block")
   (let* ((result-params (cdr (assq :result-params params)))
-	 (cmd-line (cdr (assq :cmd-line params)))
-	 (in-file (cdr (assq :in-file params)))
-	 (code-file (let ((file (org-babel-temp-file "jq-")))
-		      (with-temp-file file
-			(insert body)
-			file)))
-	 (stdin (let ((stdin (cdr (assq :stdin params))))
-		  (when stdin
-		    (let ((tmp (org-babel-temp-file "jq-stdin-"))
-			  (res (org-babel-ref-resolve stdin)))
-		      (with-temp-file tmp
-			(insert res)
-			tmp)))))
-	 (cmd (mapconcat #'identity
-			 (remq nil
-			       (list org-babel-jq-command
-				     (format "--from-file \"%s\"" code-file)
-				     cmd-line
-				     in-file))
-			 " ")))
+         (cmd-line (cdr (assq :cmd-line params)))
+         (in-file (cdr (assq :in-file params)))
+         (code-file (let ((file (org-babel-temp-file "jq-")))
+                      (with-temp-file file
+                        (insert body)
+                        file)))
+         (stdin (let ((stdin (cdr (assq :stdin params))))
+                  (when stdin
+                    (let ((tmp (org-babel-temp-file "jq-stdin-"))
+                          (res (org-babel-ref-resolve stdin)))
+                      (with-temp-file tmp
+                        (insert
+                         (cond
+                          ((stringp res) res)
+                          ((listp res) (org-babel-jq-table-to-json res))))
+                        tmp)))))
+         (cmd (mapconcat #'identity
+                         (remq nil
+                               (list org-babel-jq-command
+                                     (format "--from-file \"%s\"" code-file)
+                                     cmd-line
+                                     in-file))
+                         " ")))
     (org-babel-reassemble-table
      (let ((results
-	    (cond
-	     (stdin (with-temp-buffer
-		      (call-process-shell-command cmd stdin (current-buffer))
-		      (buffer-string)))
-	     (t (org-babel-eval cmd "")))))
+            (cond
+             (stdin (with-temp-buffer
+                      (call-process-shell-command cmd stdin (current-buffer))
+                      (buffer-string)))
+             (t (org-babel-eval cmd "")))))
        (when results
-	 (org-babel-result-cond result-params
-	   results
-	   (let ((tmp (org-babel-temp-file "jq-results-")))
-	     (with-temp-file tmp
-	       (insert results))
-	     (org-babel-import-elisp-from-file tmp)))))
+         (org-babel-result-cond result-params
+           results
+           (with-temp-buffer
+             (insert results)
+             (json-mode)
+             (json-mode-beautify)
```

Comment:
As you said, I also think jq can do this. I do use `json-mode` in `jq-interactively` to fontify the buffer, but only if the user has installed `json-mode`. I don't make it mandatory and I don't think we should when we have other options.